src/color/test/readableColor.test.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import readableColor from '../readableColor'
+
+describe('readableColor', () => {
+  it('should return black given white', () => {
+    expect(readableColor('rgb(255,255,255)')).toMatchSnapshot()
+  })
+
+  it('should return white given black', () => {
+    expect(readableColor('rgb(0,0,0)')).toMatchSnapshot()
+  })
+
+  it('should return black given white hex', () => {
+    expect(readableColor('#fff')).toMatchSnapshot()
+  })
+
+  it('should return white given black hex', () => {
+    expect(readableColor('#000')).toMatchSnapshot()
+  })
+
+  it('should return black given red', () => {
+    expect(readableColor('#FF0000')).toMatchSnapshot()
+  })
+
+  it('should return white given blue', () => {
+    expect(readableColor('#0000FF')).toMatchSnapshot()
+  })
+
+  it('should return white given this gray', () => {
+    expect(readableColor('rgb(118,118,118)')).toMatchSnapshot()
+  })
+
+  it('should return black given this gray', () => {
+    expect(readableColor('rgb(117,118,118)')).toMatchSnapshot()
+  })
+})
```

Comment:
I will add those tests!  I hadn't thought of named css.  Good one.  Hummmm.  I'll do some work on that.

The alpha colors are an issue because this function isn't applicable.  We're trying to get the most readable color given another color, but the answer for rgba/hsla depends on the backing color.  (e.g. rgba(0,0,0,0.1) in front of white should return black, but 0,0,0,0.1 in front of black should return white)

I could add an argument to the function for the backing color, but I don't think that's particularly helpful.  In most cases if you know the backing color, you should use a rgb/hex/hsl value and save the compute time.  For nearly all alpha uses, the backing color will be dynamic.  

Is it breaking with library's ethos if the function returns an error to rgba/hsla?